Source/CarthageKit/Cartfile.swift

```diff
@@ -254,7 +255,33 @@ extension Dependency: Scannable {
 			}
 		} else if scanner.scanString("git", into: nil) {
 			parser = { urlString in
-				return .success(self.git(GitURL(urlString)))
+				
+				// This is a dependency expressed as git "..."
+				let gitURL = GitURL(urlString)
+				
+				if gitURL.containsGithubDomain {
```

Comment:
These logic seems like it should be broken out into a separate function. It's quite a bit of code to put in this method and it's also a bit more general.

Maybe it could be in an initializer on `Depndency`?

```swift
extension Dependency {
  init(gitURL: GitURL) {
  }
}
```